### PR TITLE
Add SetThreadPoolSize interface

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/zilliztech/knowhere/issues/327

### CODE REVIEW

File: include/knowhere/comp/thread_pool.h [hunk 1]
```diff
@@ -89,6 +89,20 @@ class ThreadPool {
...
+    void
+    SetNumThreads(uint32_t num_threads, std::mutex& pool_mtx) {
+        if (num_threads == 0) {
+            LOG_KNOWHERE_ERROR_ << "set number of threads can not be 0";
+            return;
+        } else {
+            std::lock_guard<std::mutex> lock(pool_mtx);
+            // setNumThreads() adjust the relevant variables instead of changing the number of threads directly;
+            // If numThreads < active threads, reduce number of running threads.
+            pool_.setNumThreads(num_threads);
```
Comment (liliu-z) [inline include/knowhere/comp/thread_pool.h:101]: Is this Idempotent?

Comment (cqy123456) [reply include/knowhere/comp/thread_pool.h:100]: A std::unique_lock inside setNumThreads function. I remove this lock.

**Review by Presburger — APPROVED** [review]

File: src/common/comp/knowhere_config.cc [hunk 2]
```diff
@@ -135,12 +135,12 @@ KnowhereConfig::SetAioContextPool(size_t num_ctx) {
 
 void
 KnowhereConfig::SetBuildThreadPoolSize(size_t num_threads) {
-    knowhere::ThreadPool::InitGlobalBuildThreadPool(num_threads);
+    knowhere::ThreadPool::SetGlobalBuildThreadPoolSize(num_threads);
```
Comment (cydrain) [inline src/common/comp/knowhere_config.cc:138]: you can replace:
InitGlobalBuildThreadPool() / InitGlobalSearchThreadPool()
with:
SetGlobalBuildThreadPool() / SetGlobalSearchThreadPool()

**Pull request merged**